modules/compute/gke-node-pool/main.tf

```diff
@@ -202,6 +214,15 @@ resource "google_container_node_pool" "node_pool" {
       On the other hand, with SPECIFIC_RESERVATION you must set `specific_reservation.key` and `specific_reservation.values` to `compute.googleapis.com/reservation-name` and a list of reservation names respectively.
       EOT
     }
+    precondition {
+      condition     = var.compact_placement || var.placement_type == null || try(contains(["COMPACT"], var.placement_type), false)
+      error_message = "`COMPACT` is the only supported value for `placement_type`."
+    }
```

Comment:
Can we add another precondidtion that if var.compact_placement is true then var.placement_type should be null/empty ?

Reply:
deprecating `compact_placement` obviates this